test/arrayops.jl

```diff
@@ -2860,3 +2860,8 @@ end
     @test String(take!(b)) ==
         "BoundsError: attempt to access 2×2 Matrix{Float64} at index [10, \"bad index\"]"
 end
+
+@testset "inference of Union{T,Nothing} arrays 26771" begin
+    f(a) = (v = [1, nothing]; [v[x] for x in a])
+    @test eltype(f([1])) == Int
```

Comment:
```suggestion
    @test only(Base.return_types(f, (Int,))) === Union{Array{Int,0}, Array{Nothing,0}}
    @test only(Base.return_types(f, (UnitRange{Int},))) <: Vector
```

Reply:
The key point about #26771 is successful inference of the dimensionality in the face of uncertain eltype.